app/models/dashboard.rb

```diff
@@ -64,7 +64,10 @@ def widgets
     (JSON.parse dashboard_json)['widgets']
   end
 
-  def create_slug
-    self.slug = SlugMaker.slug(name)
+  def create_slug name, n=nil
+    self.slug = SlugMaker.slug("#{name} #{n}")
+    if !valid? && errors[:slug].any?
```

Comment:
Yeah. You could also check explicitly test whether there is a slug with that name already. Might have less side effects: `self.class.exists?(slug: slug)` or so.